pkg/providers/version/version.go

```diff
@@ -69,39 +70,39 @@ func NewDefaultProvider(kubernetesInterface kubernetes.Interface, cache *cache.C
 }
 
 func (p *DefaultProvider) Get(ctx context.Context) (string, error) {
-	var version string
-	var versionSource string
-	if version, ok := p.cache.Get(kubernetesVersionCacheKey); ok {
-		return version.(string), nil
+	if version := *p.previousVersion.Clone(); version != "" {
```

Comment:
If we validate that we never set this as an empty string (along with the assertion in operator), we can always assume this is set.